src/transport/mod.rs

```diff
@@ -12,89 +12,124 @@ pub use sending_targets_cache::{Resend, RESEND_DELAY, RESEND_MAX_ATTEMPTS};
 
 use crate::{
     quic_p2p::{Builder, EventSenders, Peer, QuicP2p, QuicP2pError, Token},
+    time::Duration,
+    timer::Timer,
     NetworkConfig,
 };
 use bytes::Bytes;
+use hex_fmt::HexFmt;
 use std::{collections::HashMap, net::SocketAddr, slice};
 
 use sending_targets_cache::SendingTargetsCache;
 
 /// Struct that handles network operations: sending and receiving messages, as well as resending on
 /// failure.
-pub struct NetworkService {
+pub struct Transport {
     quic_p2p: QuicP2p,
     cache: SendingTargetsCache,
     next_msg_token: Token,
     scheduled_messages: HashMap<u64, ScheduledMessage>,
 }
 
-impl NetworkService {
-    pub fn service_mut(&mut self) -> &mut QuicP2p {
-        &mut self.quic_p2p
+impl Transport {
+    pub fn bootstrap(&mut self) {
+        self.quic_p2p.bootstrap()
     }
 
-    pub fn next_msg_token(&mut self) -> Token {
-        self.next_msg_token = self.next_msg_token.wrapping_add(1);
-        self.next_msg_token
-    }
-
-    pub fn targets_cache_mut(&mut self) -> &mut SendingTargetsCache {
-        &mut self.cache
-    }
-
-    pub fn send_message_to_initial_targets(
+    pub fn send_message_to_targets(
         &mut self,
         conn_infos: &[SocketAddr],
         dg_size: usize,
         msg: Bytes,
-    ) -> Token {
+    ) {
+        if conn_infos.len() < dg_size {
+            warn!(
+                "Less than dg_size valid targets! dg_size = {}; targets = {:?}; msg = {:?}",
+                dg_size,
+                conn_infos,
+                HexFmt(&msg)
+            );
+        }
+
         let token = self.next_msg_token();
 
+        trace!(
+            "Sending message ID {} to {:?}",
```

Comment:
is message_id and message_token the same thing?
I think a message_id shall be consistent among the nodes, meanwhile a message_token is just a local one?

Reply:
> is message_id and message_token the same thing?

Yes, it is in our case (we don't have global message ids). Perhaps we should always refer to it as "token" and never as "id", to avoid confusion.

Reply:
right, better mention `token` only to avoid confusion.